pallets/settlement/src/lib.rs

```diff
@@ -1190,24 +1252,37 @@ impl<T: Config> Module<T> {
 
     fn ensure_instruction_validity(
         id: InstructionId,
+        is_execute: bool,
     ) -> Result<Instruction<T::Moment, T::BlockNumber>, DispatchError> {
         let details = Self::instruction_details(id);
         ensure!(
             details.status != InstructionStatus::Unknown,
             Error::<T>::UnknownInstruction
         );
-        if let (Some(trade_date), Some(value_date)) = (details.trade_date, details.value_date) {
-            ensure!(
-                value_date >= trade_date,
-                Error::<T>::InstructionDatesInvalid
-            );
-        }
-        if let SettlementType::SettleOnBlock(block_number) = details.settlement_type {
-            ensure!(
-                block_number > System::<T>::block_number(),
-                Error::<T>::InstructionSettleBlockPassed
-            );
+
+        match details.settlement_type {
+            SettlementType::SettleOnBlock(block_number) => {
+                if is_execute {
```

Comment:
Another way to do the match is:
```suggestion
        match (details.settlement_type, is_execute) {
            (SettlementType::SettleOnBlock(block_number), true) => {
                // Add the `if is_execute {` logic here.
            }
            (SettlementType::SettleOnBlock(block_number), false) => {
                // Add the `} else {` logic here.
            }
            (SettlementType::SettleManual(block_number), _) => { // Here we don't care if `is_execute` is true/false.
            }
            (_, _) => (), // Do nothing for SettleOnAffirmation.
```